### DISCUSSION TITLE
Boolean attributes compile to strings

### DISCUSSION THREAD

**Discussion by joshdcuneo** [mhanberg/temple#138] [Q&A] [answered]
Thanks for the awesome library, it feels so natural to write compared to eex. I was excited to see LiveView support.

I am having a little difficulty with boolean attributes. In eex something like works as expected. My naive temple equivalent is always disabled though because boolean is compiled to a string.

```
content_tag(:button, "Submit", disabled: false)
# <button>Submit</button>

button disabled: false, do: "Submit"
# <button disabled="false">Submit</button>
```

Is there are workaround for boolean attributes?

**Comment by joshdcuneo** [comment, marked as answer]
To answer myself, while it might be possible to handle this at compile time for a static attribute as in the example, it would not work with assigns as temple is compiled away and eex doesn't handle it either. Options seem to be use `content_tag` or disabled the button client side (with hooks in a LiveView if needed).

**Comment by mhanberg** [comment]
This is a convenience I'd like to support. 

Currently in my app I have a very hacky function to help with this. 

```elixir
fieldset merge_if([], [disabled: true], not is_nil(@foo.finished_at)) do
  # ...
end

def merge_if(keyword_1, keyword_2, condition) do
  if condition do
    Keyword.merge(keyword_1, keyword_2)
  else
    keyword_1
  end
end
```

**Reply by eahanson** [reply]
~~It looks like Phoenix _always_ drops attrs whose value is `false` or `nil` [here in Tag.ex](https://github.com/phoenixframework/phoenix_html/blob/c7201bad056bc9952e0cd9e6bd47e6285606019e/lib/phoenix_html/tag.ex#L133). Would that work in Temple too, or are there situations where that would be unwelcome?~~ Nevermind, I mis-read joshdcuneo's comment. If it doesn't work with assigns, it probably wouldn't be a useful addition.

**Comment by mhanberg** [comment]
I've opened a PR to address this #139

**Comment by mhanberg** [comment]
I've just released [v0.7.0](https://github.com/mhanberg/temple/releases/tag/v0.7.0) that includes this functionality.
